### DISCUSSION TITLE
Better integration of Transform struct

### DISCUSSION THREAD

**Discussion by Boyfinn** [raysan5/raylib#3670] [Ideas]
So far, 3D models have support for the Transform-type. But it would be great if this could be better integrated into rmodels for better consistency when handling 3D transforms and drawing 3D models.
Also, more helper functions for working with transforms, such as converting a Transform into a 4x4 transformation matrix like so:
```
static Matrix Transform2Matrix(Transform Trs)
{
	//Temporary transform matrix
	Matrix _tfMatrix = MatrixIdentity();	//transformMatrix

	//Matrix scale (in local space)
	Matrix _scMatrix = MatrixScale(Trs.scale.x, Trs.scale.y, Trs.scale.z); //scaleMatrix
	_tfMatrix = MatrixMultiply(_tfMatrix, _scMatrix);	//applyScale

	//Convert Quaternion to rotation matrix
	Matrix _rtMatrix = QuaternionToMatrix(Trs.rotation);	//rotationMatrix

	//matrix rotation
	Matrix _trMatrix = MatrixIdentity();	//translationMatrix
	if (Trs.translation.x != 0 || Trs.translation.y != 0 || Trs.translation.z != 0)
		_trMatrix = MatrixTranslate(Trs.translation.x, Trs.translation.y, Trs.translation.z);

	//Finalize transform
	_tfMatrix = MatrixMultiply(_tfMatrix, _rtMatrix);	//applyRotation
	_tfMatrix = MatrixMultiply(_tfMatrix, _trMatrix);	//applyTransform

	//this->_model.transform = _tfMatrix;
	return _tfMatrix;
}
```
And vice-versa:
```
static Vector3 Matrix2Vec3(Matrix Mat)
{
	Vector3 _tr = Vector3Zero();
	_tr.x = Mat.m12;
	_tr.y = Mat.m13;
	_tr.z = Mat.m14;
	return _tr;
}
```

**Comment by Boyfinn** [comment]
@raysan5 Also, note that in the documentation the comment for the Transform struct has a typo:
// Transform, **vectex** transformation data

**Reply by raysan5** [reply]
It's fixed on latest raylib.h from github master branch.

**Reply by Boyfinn** [reply]
The cheatsheet on the main page still has this typo.

**Comment by raysan5** [comment]
`struct Transform` does not belong to `raymath.h`, so I prefer to keep functions separated but feel free to proposed a list of useful functions, maybe they can be added to a separate library for users requiring them.

**Reply by Boyfinn** [reply]
How about better integrating this into rmodels drawing functions? A transform structure is such an integral part of 3D graphics, But I don't see it being used in the library much at all. I think it would be easier and more flexible to have these incorporate better into the Model type, It's Matrix transform functions and any drawing related 3D function in general.

**Reply by raysan5** [reply]
I'm afraid it would require redesigning multiple parts of `rmodels` module to accomodate those changes and probably breaking raylib API  and compatibility with previous versions. I'm afraid this is out of scope for now.

**Reply by Boyfinn** [reply]
That's alright. I might fork rmodels at some point to implement these features if that's ok with you?

**Reply by raysan5** [reply]
Sure! Feel free to create a fork!
